### DISCUSSION TITLE
Incorrect FCnt start value?

### DISCUSSION THREAD

**Discussion by sanijs** [Lora-net/LoRaMac-node#1571] [Q&A]
It has previous been mentioned that uplink FC starts with 1 instead of 0 here https://github.com/Lora-net/LoRaMac-node/issues/890

When reading the LoRaWAN specification 1.0.4 it seems that it has been updated and clarified that the FCnt should start with 0 after a successful join procedure.

Is it correctly understood that the FCnt should increment after each uplink, so the first uplink FCnt should be 0 and not 1?

**Comment by mluis1** [comment]
We believe that the `FCntUp` management is correct and compliant with LoRaWAN 1.0.4 specification or any other LoRaWAN specification version.

![image](https://github.com/Lora-net/LoRaMac-node/assets/5763295/48b8687b-0f30-4b37-b07a-b6a573a3e2f9)

The specification says that after receiving the `JoinAccept` the `FCnt` variables value must be reset to 0. (First highlighted sentence)
Then it says that every new uplink must use an `FCnt` value incremented by 1. (Second highlighted sentence)
The specification also says that `FCnt` values equal to 0 must be accepted which makes this an exception. (Third highlighted sentence)

Therefore according to the specification we believe that a first uplink using an `FCntUp=0` or `FCntUp=1` are accepted behaviors.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.
